### ISSUE TITLE
Boolean condition with non entity member in where() throw exception

### ISSUE THREAD

**Issue by cdie** [aspnet/EntityFramework#4868]
I don't know if this fix https://github.com/aspnet/EntityFramework/pull/4141 solves this problem, but I cannot make boolean test within a LINQ Where function with a non-entity member test.

In my example, I have a ViewModel that contains a simple flag.

In my controller method, I want to change the Where condition depending of this flag :

`var postes = _context.Set<GestPoste>().Where(m => m.Ps == guid && (viewmodel.IsTla == true ? m.Range >= 800 : m.Range < 800)).ToList();
`
This will give this SQL Query :

```
exec sp_executesql N'SELECT [m].[PERIPH_GUID], [m].[NAME], [m].[PS_GUID], [m].[RANGE]
FROM [GEST_POSTE] AS [m]
WHERE ([m].[PS_GUID] = @__guid_0) AND CASE
    WHEN @__model_IsTla_1
    THEN [m].[RANGE] >= 800 ELSE [m].[RANGE] < 800
END',N'@__guid_0 uniqueidentifier,@__model_IsTla_1 bit',@__guid_0='F6AC31E2-A2BB-403B-8D4D-264AD3A306B5',@__model_IsTla_1=1
```

Which, of course, gives an error.

My only workaround today is 

```
Func<GestPoste, bool> fun;
if (model.IsTla)
   fun = (m) => m.Range >= 800;
else
    fun = (m) => m.Range < 800;
var postes = _context.Set<GestPoste>().Where(m => m.Ps == guid && fun(m)).ToList();
```

Is this a bug ?

**Comment by smitpatel** [comment]
The fix for #4141 fixes above partially. Specifically this is how the where clause would look like
```
WHERE ([m].[PS_GUID] = @__guid_0) AND CASE
    WHEN @__model_IsTla_1 = 1
    THEN [m].[RANGE] >= 800 ELSE [m].[RANGE] < 800
END
```
It would be still incorrect syntax. While fixing #4141 I hit the same issue. I couldn't find exact words in the documentation but from what I have tried, it is not possible that result of case block is a Boolean condition.
If you can translate the query in T-Sql then please provide it.

**Comment by smitpatel** [comment]
Found T-Sql though it is very long expansion. It can be reduced using logical operators. Looking at the fixing it.
```
WHERE ([m].[PS_GUID] = @__guid_0) AND 
(CASE
    WHEN @__model_IsTla_1 = 1
    THEN 
		CASE
			WHEN [m].[RANGE] >= 800
			THEN 1
			ELSE 0
		END
	ELSE
		CASE
			WHEN [m].[RANGE] < 800
			THEN 1
			ELSE 0
		END
	END) = 1
```
